notebooks/assistants/function_calling/assistants_function_calling_with_bing_search.ipynb

```diff
@@ -0,0 +1,439 @@
+      "Requirement already satisfied: httpcore==1.* in c:\\users\\hakesava\\appdata\\local\\anaconda3\\envs\\trial\\lib\\site-packages (from httpx<1,>=0.23.0->openai) (1.0.2)\n",
+      "Requirement already satisfied: h11<0.15,>=0.13 in c:\\users\\hakesava\\appdata\\local\\anaconda3\\envs\\trial\\lib\\site-packages (from httpcore==1.*->httpx<1,>=0.23.0->openai) (0.14.0)\n",
+      "Requirement already satisfied: annotated-types>=0.4.0 in c:\\users\\hakesava\\appdata\\local\\anaconda3\\envs\\trial\\lib\\site-packages (from pydantic<3,>=1.9.0->openai) (0.6.0)\n",
+      "Requirement already satisfied: pydantic-core==2.16.1 in c:\\users\\hakesava\\appdata\\local\\anaconda3\\envs\\trial\\lib\\site-packages (from pydantic<3,>=1.9.0->openai) (2.16.1)\n",
+      "Requirement already satisfied: colorama in c:\\users\\hakesava\\appdata\\local\\anaconda3\\envs\\trial\\lib\\site-packages (from tqdm>4->openai) (0.4.6)\n",
+      "Note: you may need to restart the kernel to use updated packages.\n"
+     ]
+    }
+   ],
+   "source": [
+    "#Install the packages\\\n",
+    "%pip install requests openai"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Resource config setup\n",
+    "Update the following config to include details of your Azure OpenAI and Bing Search resources"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 41,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "config = {\n",
+    "    \"DEPLOYMENT_NAME\":\"<DEPLOYMENT_NAME>\",\n",
+    "    \"AOAI_API_BASE\":\"https://<YOUR_RESOURCE_NAME>.openai.azure.com\",\n",
+    "    \"AOAI_API_VERSION\":\"2024-02-15-preview\",\n",
+    "\n",
+    "    \"AOAI_API_KEY\":\"<AOAI_RESOURCE_API_KEY>\",\n",
+    "    \"BING_SEARCH_SUBSCRIPTION_KEY\": \"<BING_SEARCH_SUBSCRIPTION_KEY>\"\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 42,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import requests\n",
+    "import json\n",
+    "import time \n",
+    "\n",
+    "from openai import AzureOpenAI\n",
+    "\n",
+    "azure_endpoint = config[\"AOAI_API_BASE\"]\n",
+    "api_version = config[\"AOAI_API_VERSION\"]\n",
+    "aoai_api_key= config[\"AOAI_API_KEY\"] \n",
+    "deployment_name = config['DEPLOYMENT_NAME']  \n",
+    "\n",
+    "bing_search_subscription_key = config['BING_SEARCH_SUBSCRIPTION_KEY']\n",
+    "bing_search_url = \"https://api.bing.microsoft.com/v7.0/search\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define a function to call the Bing Search APIs\n",
+    "To learn more about using the Bing Search APIs with Azure OpenAI, see [Bing Search APIs, with your LLM](https://learn.microsoft.com/bing/search-apis/bing-web-search/use-display-requirements-llm)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 43,
+   "metadata": {},
+   "outputs": [],
+   "source": [
```

Comment:
Moves user specified values into a "[parameters block](https://papermill.readthedocs.io/en/latest/usage-parameterize.html)" per the template.

```suggestion
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Parameters\n",
    "Update the following config to include details of your Azure OpenAI and Bing Search resources"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "tags": [
     "parameters"
    ]
   },
   "outputs": [],
   "source": [
    "deployment_name = \"<DEPLOYMENT_NAME>\"\n",
    "aoai_api_base = \"https://<YOUR_RESOURCE_NAME>.openai.azure.com\"\n",
    "aoai_api_version = \"2024-02-15-preview\"\n",
    "aoai_api_key = \"<AOAI_RESOURCE_API_KEY>\"\n",
    "bing_search_subscription_key = \"<BING_SEARCH_SUBSCRIPTION_KEY>\"\n",
    "bing_search_url = \"https://api.bing.microsoft.com/v7.0/search\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Define a function to call the Bing Search APIs\n",
    "To learn more about using the Bing Search APIs with Azure OpenAI, see [Bing Search APIs, with your LLM](https://learn.microsoft.com/bing/search-apis/bing-web-search/use-display-requirements-llm)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "import json\n",
    "import time\n",
    "\n",
    "from openai import AzureOpenAI\n",
    "\n",
    "\n",

```

Reply:
I've tried doing this, the pre-commit checks erase it/fail.